packages/template-retail-react-app/app/components/store-locator-modal/store-locator-content.jsx

```diff
@@ -16,11 +16,56 @@ import {
 } from '@salesforce/retail-react-app/app/components/shared/ui'
 import StoresList from '@salesforce/retail-react-app/app/components/store-locator-modal/stores-list'
 import StoreLocatorInput from '@salesforce/retail-react-app/app/components/store-locator-modal/store-locator-input'
-import {STORE_LOCATOR_DISTANCE} from '@salesforce/retail-react-app/app/constants'
+import {
+    SUPPORTED_STORE_LOCATOR_COUNTRIES,
+    DEFAULT_STORE_LOCATOR_COUNTRY,
+    DEFAULT_STORE_LOCATOR_POSTAL_CODE
+} from '@salesforce/retail-react-app/app/constants'
 
-const StoreLocatorContent = ({form, submitForm, storesInfo, searchStoresParams}) => {
+const StoreLocatorContent = ({
+    form,
+    submitForm,
+    storesInfo,
+    searchStoresParams,
+    setSearchStoresParams,
+    userHasSetManualGeolocation,
+    setUserHasSetManualGeolocation
+}) => {
     const intl = useIntl()
 
+    const getGeolocationError = () => {
+        console.log('Unable to retrieve your location')
+        setSearchStoresParams({
+            countryCode: DEFAULT_STORE_LOCATOR_COUNTRY.countryCode,
+            postalCode: DEFAULT_STORE_LOCATOR_POSTAL_CODE,
+            limit: 10
+        })
+    }
+
+    const getGeolocationSuccess = (position) => {
+        const latitude = position.coords.latitude
+        const longitude = position.coords.longitude
+        setSearchStoresParams({
+            latitude: position.coords.latitude,
+            longitude: position.coords.longitude,
+            limit: 10
+        })
+        console.log(`Latitude: ${latitude}, Longitude: ${longitude}`)
+    }
+
+    const getUserGeolocation = () => {
+        if (typeof navigator !== 'undefined' && navigator.geolocation) {
+            navigator.geolocation.getCurrentPosition(getGeolocationSuccess, getGeolocationError)
+            setUserHasSetManualGeolocation(false)
+        } else {
+            console.log('Geolocation not supported')
+        }
+    }
+
+    useEffect(() => {
+        if (!userHasSetManualGeolocation) getUserGeolocation()
+    }, [])
```

Comment:
have you considered creating a `useGeolocation` hook to reduce the complexity of `StoreLocatorContent` component

Reply:
That would be neat for sure. But in my very initial SPIKE for this feature and Epic, I received recommendation from @bendvc to reduce the number of dependencies that come out of the components folder to other areas of the app as much as possible to ease the future workload for extensibility, so I tried to not make exported functions in other files.

Reply:
Ok, if that's a decision you have made for not creating files. How about creating a function inside this file? 

```js
const useGeolocation = () => {}

const StoreLocatorContent = () => {
   const location = useGeolocation()
}
```